docs/audio.rst

```diff
@@ -83,8 +86,216 @@ The built-in sounds can be called using ``audio.play(Sound.NAME)``.
+            audio.play(Sound.TWINKLE, wait=False)
+            display.show(Image.BUTTERFLY)
+
+        sleep(500)
+        display.clear()
+
+
+Sound Effects **V2**
+====================
+
+.. py:class::
+    SoundEffect(preset=None, freq_start=500, freq_end=2500, duration=500, vol_start=255, vol_end=0, wave=WAVE_SQUARE, fx=None, interpolation=INTER_LOG)
+
+    An ``SoundEffect`` instance represents a sound effect, composed by a set of
+    parameters configured via the constructor or attributes.
+
+    All the parameters are optional, with default values as shown above, and
+    they can all be modified via attributes of the same name. For example, we
+    can first create an effect ``my_effect = SoundEffect(duration=1000)``,
+    and then change its attributes ``my_effect.duration = 500``.
+
+    :param preset: An existing SoundEffect instance to use as a base, its values
+        are cloned in the new instance, and any additional arguments provided
+        overwrite the base values.
+    :param freq_start: Start Frequency in Hertz (Hz), eg: ``400``
+    :param freq_end: End Frequency in Hertz (Hz), eg: ``2000``
+    :param duration: Duration of the sound (ms), eg: ``500``
+    :param vol_start: Start volume value, range 0-255, eg: ``120``
+    :param vol_end: End volume value, range 0-255, eg: ``255``
+    :param wave: Type of wave shape, one of these values: ``WAVE_SINE``,
+        ``WAVE_SAWTOOTH``, ``WAVE_TRIANGLE``, ``WAVE_SQUARE``,
+        ``WAVE_NOISE`` (randomly generated noise).
+    :param fx: Effect to add on the sound, one of the following values:
+        ``FX_TREMOLO``, ``FX_VIBRATO``, ``FX_WARBLE``, or ``None``.
+    :param interpolation: The type of curve between the start and end
+        frequencies, different wave shapes have different rates of change
+        in frequency. One of the following values: ``INTER_LINEAR``,
+        ``INTER_CURVE``, ``INTER_LOG``.
+
+    .. py:attribute:: freq_start
+
+        Start Frequency in Hertz (Hz)
+
+    .. py:attribute:: freq_end
+
+        End Frequency in Hertz (Hz)
+
+    .. py:attribute:: duration
+
+        Duration of the sound (ms), eg: ``500``
+
+    .. py:attribute:: vol_start
+
+        Start volume value, range 0-255, eg: ``120``
+
+    .. py:attribute:: vol_end
+
+        End volume value, range 0-255, eg: ``255``
+
+    .. py:attribute:: wave
+
+        Type of wave shape, one of these values: ``WAVE_SINE``,
+        ``WAVE_SAWTOOTH``, ``WAVE_TRIANGLE``, ``WAVE_SQUARE``,
+        ``WAVE_NOISE`` (randomly generated noise).
+
+    .. py:attribute:: fx
+
+        Effect to add on the sound, one of the following values:
+        ``FX_TREMOLO``, ``FX_VIBRATO``, ``FX_WARBLE``, or ``None``.
+
+    .. py:attribute:: interpolation
+
+        The type of curve between the start and end
+        frequencies, different wave shapes have different rates of change
+        in frequency. One of the following values: ``INTER_LINEAR``,
+        ``INTER_CURVE``, ``INTER_LOG``.
```

Comment:
Creating a comment here to be able to keep this discussion thread in parallel.

From the PR description:
> The parameter name `interpolation` is too long, we are still looking for alternatives.
> * `curve` is being considered, but one of the options is actually `INTER_CURVE`, so we might have to rename that to something else
> * `ramp`?

Reply:
Adding @dpgeorge's comment from https://github.com/bbcmicrobit/micropython/pull/753#issuecomment-1179641170 here as well:
> > The parameter name `interpolation` is too long, we are still looking for alternatives
> 
> How about `shape`? That matches the internal CODAL name.

Reply:
`shape` could work! It's probably better than `ramp` as well, although maybe it's a matter of opinion.

@microbit-giles For the MicroPython SoundEffects, what are your thoughts about changing the name of `interpolation` (long and not a well known word for young coders) to `shape`?